pkg/server/public/tests/form_test.go

```diff
@@ -0,0 +1,189 @@
+		return
+	}
+
+	tcs := []struct {
+		name   string
+		fields []*types.FieldDefinition
+	}{
+		{
+			name: "With Text Field",
+			fields: tu.Fields(
+				tu.ATextField(tu.FieldName("My Field")),
+			),
+		}, {
+			name: "With Required Text Field",
+			fields: tu.Fields(
+				tu.ATextField(tu.FieldName("My Field"), tu.FieldRequired(true)),
+			),
+		}, {
+			name: "With Key Text Field",
+			fields: tu.Fields(
+				tu.ATextField(tu.FieldName("My Field"), tu.FieldRequired(true), tu.FieldKey(true)),
+			),
+		}, {
+			name: "With Multiline Field",
+			fields: tu.Fields(
+				tu.AMultilineTextField(tu.FieldName("My Field")),
+			),
+		}, {
+			name: "With Required Multiline Field",
+			fields: tu.Fields(
+				tu.AMultilineTextField(tu.FieldName("My Field"), tu.FieldRequired(true)),
+			),
+		}, {
+			name: "With Quantity Field",
+			fields: tu.Fields(
+				tu.AQuantityField(tu.FieldName("My Field")),
+			),
+		}, {
+			name: "With Required Quantity Field",
+			fields: tu.Fields(
+				tu.AQuantityField(tu.FieldName("My Field"), tu.FieldRequired(true)),
+			),
+		}, {
+			name: "With Key Quantity Field",
+			fields: tu.Fields(
+				tu.AQuantityField(tu.FieldName("My Field"), tu.FieldRequired(true), tu.FieldKey(true)),
+			),
+		}, {
+			name: "With Month Field",
+			fields: tu.Fields(
+				tu.AMonthField(tu.FieldName("My Field")),
+			),
+		}, {
+			name: "With Required Month Field",
+			fields: tu.Fields(
+				tu.AMonthField(tu.FieldName("My Field"), tu.FieldRequired(true)),
+			),
+		}, {
+			name: "With Key Month Field",
+			fields: tu.Fields(
+				tu.AMonthField(tu.FieldName("My Field"), tu.FieldRequired(true), tu.FieldKey(true)),
+			),
+		}, {
+			name: "With Date Field",
+			fields: tu.Fields(
+				tu.ADateField(tu.FieldName("My Field")),
+			),
+		}, {
+			name: "With Required Date Field",
+			fields: tu.Fields(
+				tu.ADateField(tu.FieldName("My Field"), tu.FieldRequired(true)),
+			),
+		}, {
+			name: "With Key Date Field",
+			fields: tu.Fields(
+				tu.ADateField(tu.FieldName("My Field"), tu.FieldRequired(true), tu.FieldKey(true)),
+			),
+		}, {
+			name: "With Week Field",
+			fields: tu.Fields(
+				tu.AWeekField(tu.FieldName("My Field")),
+			),
+		}, {
+			name: "With Required Week Field",
+			fields: tu.Fields(
+				tu.AWeekField(tu.FieldName("My Field"), tu.FieldRequired(true)),
+			),
+		}, {
+			name: "With Key Week Field",
+			fields: tu.Fields(
+				tu.AWeekField(tu.FieldName("My Field"), tu.FieldRequired(true), tu.FieldKey(true)),
+			),
+		}, {
+			name: "With Reference Field",
+			fields: tu.Fields(
+				tu.AReferenceField(otherForm, tu.FieldName("My Field")),
+			),
+		}, {
+			name: "With Required Reference Field",
+			fields: tu.Fields(
+				tu.AReferenceField(otherForm, tu.FieldName("My Field"), tu.FieldRequired(true)),
+			),
+		}, {
+			name: "With Key Reference Field",
+			fields: tu.Fields(
+				tu.AReferenceField(otherForm, tu.FieldName("My Field"), tu.FieldRequired(true), tu.FieldKey(true)),
+			),
```

Comment:
Add cases for broken inputs. i.e.: missing required fields, wrong types, etc

Reply:
Yeah... this part was just to test the rest api, make sure that we can read and write without breaking anything. Def. would need more work in the future, more cases, etc.